KioskApplication/src/main/java/controller/ScreenController.java

```diff
@@ -44,7 +45,7 @@ protected MapController getMapController() {
         return view;
     }
 
-    public abstract void onMapLocationClicked(Point2D location);
+    public abstract void onMapLocationClicked(javafx.scene.input.MouseEvent e, Point2D location);
```

Comment:
This isn't great. Let's just pass the MouseEvent instead of the location if we need to.

Reply:
Since the location is stored within the MouseEvent

Reply:
MouseEvent need to be passed in to detect a double-click event